static/src/js/admin/field/select.js

```diff
@@ -15,40 +15,51 @@ define(['jquery-nos'], function($) {
      */
     return function ($field, options, is_new, is_expert) {
         var $choices = getFieldProperty($field, 'field_choices');
+        var $mandatory = getFieldProperty($field, 'field_mandatory');
+        var $default_value = getFieldProperty($field, 'field_default_value');
 
         // Updates the preview on choices change
         $choices.on('blur change keyup', generateFieldDefaultValue);
+        $mandatory.on('change', generateFieldDefaultValue);
 
         generateFieldDefaultValue();
 
         function generateFieldDefaultValue() {
             // Gets the default value
-            var $default_value = getFieldProperty($field, 'field_default_value');
-            var default_value_value = $default_value.val();
+            var default_value_value = $default_value.val().toString();
 
             // Creates the select
             var $new = $('<select />').attr({
                 name: $default_value.attr('name'),
                 id: $default_value.attr('id')
             });
 
+            if (!$mandatory.prop('checked')) {
+                $new.append(
+                    $('<option value="" ' + (default_value_value === "" ? 'selected' : '') + '></option>')
```

Comment:
You should check if there isn't already an empty option before injecting one, otherwise you could have a double empty option.

Reply:
I don't think this would be coherent. The added option represents the absence of value (`null`), while a manually added option is still actually an option (non-`null` value). And this script is necessary to allow the default value of the select to be `null` when the field is optional.

Reply:
What would not be coherent is that after merging this PR, all the select fields with an empty option and the "Mandatory" field checked will have a double empty selected value.